### PR TITLE
Introduce state query service

### LINKED ISSUE #7751: Introduce a service to query the engine state for process id
**Description**

Introduce a QueryService to `io.camunda.zeebe.engine.state` that implements 3 simple methods to retrieve the BpmnProcessId for a key:

- getBpmnProcessIdForProcess(Long key)
- getBpmnProcessIdForProcessInstance(Long key)
- getBpmnProcessIdForJob(Long key)

Make the QueryService available to other classes by adding it as a deprecated parameter to the PartitionListener interface. It needs access to the ZeebeState, which it can easily get from the place where the partition listener interface is called, see dbc5cc1 as example

### PR DESCRIPTION
**Pull request by korthout**
## Description

Introduces a new query service interface and implementation to the engine's state module, that can be used to find the bpmn process id of a process, process instance or job.

The interface is deprecated to avoid additional usages.

## Related issues

closes #7751 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by korthout** [comment]
@npepinpe please review

@saig0 please also have a look since this relates to the engine

**Comment by saig0** [comment]
@korthout looks good from my side :+1: Nice and clean. Well done :1st_place_medal:

File: engine/src/main/java/io/camunda/zeebe/engine/state/query/StateQueryService.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import io.camunda.zeebe.engine.state.immutable.JobState;
+import io.camunda.zeebe.engine.state.immutable.ProcessState;
+import io.camunda.zeebe.engine.state.immutable.ZeebeState;
+import io.camunda.zeebe.engine.state.instance.ElementInstance;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceRecord;
+import java.util.Optional;
+import org.agrona.DirectBuffer;
+
+public class StateQueryService implements QueryService {
```
Comment (npepinpe) [inline engine/src/main/java/io/camunda/zeebe/engine/state/query/StateQueryService.java:25]:
```suggestion
public final class StateQueryService implements QueryService {
```

File: engine/src/test/java/io/camunda/zeebe/engine/state/query/StateQueryServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ZeebeStateExtension.class)
+class StateQueryServiceTest {
```
Comment (npepinpe) [inline engine/src/test/java/io/camunda/zeebe/engine/state/query/StateQueryServiceTest.java:28]:
```suggestion
final class StateQueryServiceTest {
```

File: engine/src/test/java/io/camunda/zeebe/engine/state/query/StateQueryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+  private StateQueryService sut;
```
Comment (npepinpe) [inline engine/src/test/java/io/camunda/zeebe/engine/state/query/StateQueryServiceTest.java:30]: :thought_balloon: I see Peter's influence is spreading :smile:

**Review by npepinpe — APPROVED** [review]
:rocket: 

- :+1: Looks great, nothing to change :smile: 
- :+1: Thank you for always documenting public interfaces :heart_eyes: 
- :upside_down_face: It seems `Records` is used purely for testing, so the `util` module is probably the wrong place for it. I know it's a little out of scope, however, but it might be better placed with other engine test utilities. I would be fine with doing this here as a separate commit, but we can also do that in a different PR.

File: engine/src/test/java/io/camunda/zeebe/engine/state/query/StateQueryServiceTest.java (hunk 3 above)
Comment (korthout) [reply engine/src/test/java/io/camunda/zeebe/engine/state/query/StateQueryServiceTest.java:30]: Haha, I was actually already used to this before 😄

**Comment by korthout** [comment]
> It seems `Records` is used purely for testing, so the `util` module is probably the wrong place for it.

@npepinpe I'm not sure what you mean. It is not in the `util` module, it's in a util package of the `engine` module. Perhaps I'm missing something.

**Comment by npepinpe** [comment]
Oops, my bad 🙈

**Comment by korthout** [comment]
No worries 😄 

I guess, this can be merged then

bors merge

**Comment by korthout** [comment]
> Build failed:
> 
> * [continuous-integration/jenkins/branch](https://ci.zeebe.camunda.cloud/job/camunda-cloud/job/zeebe/job/staging/978/display/redirect)

Some build hick-up or so
> ```
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.1.2:resolve-plugins (go-offline) on project zeebe-bpmn-model: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at org.apache.maven.plugins:maven-resources-plugin:jar:3.2.0 -> org.apache.maven.shared:maven-filtering:jar:3.2.0 -> org.sonatype.plexus:plexus-build-api:jar:0.0.7: Failed to read artifact descriptor for org.sonatype.plexus:plexus-build-api:jar:0.0.7: Could not transfer artifact org.sonatype.plexus:plexus-build-api:pom:0.0.7 from/to camunda-nexus (http://repository-ci-camunda-cloud.nexus:8081/content/groups/internal/): /home/jenkins/agent/workspace/camunda-cloud_zeebe_staging/m2-repository/org/sonatype/plexus/plexus-build-api/0.0.7/plexus-build-api-0.0.7.pom.part (No such file or directory) -> [Help 1]
> ```
> 

bors merge

**Comment by korthout** [comment]
> Build failed:
> 
> * [continuous-integration/jenkins/branch](https://ci.zeebe.camunda.cloud/job/camunda-cloud/job/zeebe/job/staging/979/display/redirect)

Ugh

bors retry

**Pull request merged**